pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractor.java

```diff
@@ -74,7 +74,7 @@ public GenericRow extract(GenericRecord from, GenericRow to) {
       }
     } else {
       for (String fieldName : _fields) {
-        Object value = from.get(fieldName);
+        Object value = from.hasField(fieldName) ? from.get(fieldName) : null;
```

Comment:
This will introduce one extra map lookup per field. If we want to avoid this, we can change the `T` to `GenericData.Record` and make a single lookup

Reply:
It is true that this introduces an extra lookup. In Venice, we have maintained the single lookup like this: https://github.com/linkedin/venice/blob/main/internal/venice-client-common/src/main/java/com/linkedin/venice/compute/ComputeUtils.java#L382

In the context of the Pinot code here, the solution could look like this:

```
Field field = from.getSchema().getField(fieldName);
Object value = field == null ? null : from.get(field.pos());
```

This works on `GenericRecord`. No need to cast or change the type.

Reply:
@FelixGV I don't see an API in `GenericRecord` to lookup an index. Do you use an even higher version Avro?

Reply:
Sorry my bad. It is in the `IndexedRecord`. Yes this is the preferred solution!

Reply:
Although hashmap lookup is O(1), using field.pos() is definitely faster and avoids that lookup. Updated the code.

Reply:
> we can change the T to GenericData.Record and make a single lookup

@Jackie-Jiang I didn't quite get this. Can you elaborate?

Reply:
> > we can change the T to GenericData.Record and make a single lookup
> 
> @Jackie-Jiang I didn't quite get this. Can you elaborate?

This is no longer relevant. I missed the fact that `GenericRecord` has an API to lookup an index